src/http/reply.cc

```diff
@@ -98,20 +99,27 @@ static const std::unordered_map<reply::status_type, std::string_view> status_str
     {reply::status_type::network_read_timeout, "598 Network Read Timeout"},
     {reply::status_type::network_connect_timeout, "599 Network Connect Timeout"}};
 
-static const std::string_view& to_string_view(reply::status_type status) {
-  if (auto found = status_strings.find(status); found != status_strings.end()) [[likely]]
-    return found->second;
-  return status_strings.at(reply::status_type::internal_server_error);
+template<typename Func>
+static auto with_string_view(reply::status_type status, Func&& func) -> std::invoke_result_t<Func, std::string_view> {
+  if (auto found = status_strings.find(status); found != status_strings.end()) [[likely]] {
+     return func(found->second);
+  }
+  auto dummy_buf = std::to_string(int(status));
```

Comment:
Patch comment says
> jumping through hoops to avoid creating string objects
but std::to_string does create a string object, so what's the point?

Reply:
Invoking func() on map data does not, thus common case still only touches static data, instead of creating a temporary.